test/sport-router-test.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+//require in the mock env
+require('./mock-env.js');
+
+//require npm modules
+const expect = require('chai').expect;
+const superagent = require('superagent');
+
+//require app modules
+const Sport = require('../model/sport.js');
+
+//start server
+require('../server.js');
+
+//create test vars
+const baseURL = `http://localhost:${process.env.PORT}`;
+
+describe('testing sport router', function(){
+  afterEach((done) => {
+    Sport.remove({})
+    .then(() => done())
+    .catch(done);
+  });
+  describe('testing POST /api/sports', function(){
+    it('should return a sport', (done) => {
+      superagent.post(`${baseURL}/api/sports`)
+      .send({sport: 'football', announcer: 'john lynch', network: 'fox'})
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.announcer).to.equal('john lynch');
+        expect(res.body.network).to.equal('fox');
+        expect(res.body.sport).to.equal('football');
+        expect(Boolean(res.body.created)).to.equal(true);
+        done();
+      })
+      .catch(done);
+    });
+
+    it('should return 400 status', (done) => {
+      superagent.post(`${baseURL}/api/sports`)
+      .send({})
+      .then(done)
+      .catch(err => {
+        expect(err.status).to.equal(400);
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('testing GET /api/sports/:id', function(){
+    beforeEach(done => {
+      new Sport({sport: 'football', announcer: 'john lynch', network: 'fox'}).save()
+        .then(sport => {
+          this.tempSport = sport;
+          done();
+        })
+        .catch(done);
+    });
+    it('should return a sport', (done) => {
+      console.log(this.tempSport);
+      superagent.get(`${baseURL}/api/sports/${this.tempSport._id}`)
+        .then(res => {
+          expect(res.status).to.equal(200);
+          expect(res.body._id).to.equal(this.tempSport._id.toString());
+          expect(res.body.sport).to.equal('football');
+          expect(res.body.announcer).to.equal('john lynch');
+          expect(res.body.network).to.equal('fox');
+          expect(Boolean(res.body.created)).to.equal(true);
+          done();
+        })
+        .catch(done);
+    });
+
+    it('should return a 404 due to bad id', (done) => {
+      superagent.get(`${baseURL}/api/sports/34343`)
+        .then(done)
+        .catch(err => {
+          expect(err.status).to.equal(404);
+          done();
+        })
+        .catch(done);
+    });
+  });
+});
+
+
+
+describe('testing DELETE /api/sports', function(){
+  beforeEach(done => {
```

Comment:
The reason your commented out test for the ``get all IDs`` route was not passing was because your database already had a Sport in there that did not get cleaned up due to this ``beforeEach`` block, so it was essentially the first item in the array the whole time and did not match with the ``tempSport._id`` you were comparing it to.

``beforeEach`` means whatever functions you put in that block, it will run **before each test** within the scope of its describe block. 

So for your delete route tests, it would create a a sport, and then get deleted successfully by your first ``DELETE`` test that expects a status 204 for successful deletion.

But when it reaches the next ``DELETE`` test that expects 404 (and therefore, doesn't actually require that any mock data be created for it), the ``beforeEach`` block creates *another* sport that goes into the Mongo database and never gets cleaned up, so it conflicts with your statement down on Line 137 ``expect(res.body[0]._id).to.equal(this.tempSport._id.toString());``. 

There are at least two ways to solve this problem which I will let you figure out on your own. :) See me if you need help.